lib/chef/resource/homebrew_install.rb

```diff
@@ -0,0 +1,148 @@
+      end
+      ```
+      DOC
+
+      property :xcode_tools_url, String,
+        description: "A url pointing to a 'Command Line Tools for Xcode' dmg file"
+
+      property :xcode_tools_pkg_name, String,
+        description: "The name of the pkg inside the dmg located at the xcode_tools_url"
+
+      property :brew_source_url, String,
+        description: "A url pointing to a Homebrew installer",
+        default: "https://codeload.github.com/Homebrew/brew/zip/master"
+
+      property :user, String,
+        description: "The user to install Homebrew as. Note: Homebrew cannot be installed as root.",
+        required: true
+
+      action :install do
+        # Avoid all the work in the below resources if homebrew is already installed
+        return if ::File.exist?("/usr/local/bin/brew")
+
+        # check if 'user' is root and raise an exception if so
+        if Etc.getpwnam(new_resource.user).uid == 0
+          msg = "You are attempting to install Homebrew as Root. This is not permitted by Homebrew. Please run this as a standard user with admin rights"
+          raise Chef::Exceptions::InsufficientPermissions, msg
+        end
+
+        USER_HOME = Dir.home(new_resource.user).freeze
+        HOMEBREW_CACHE = "#{USER_HOME}/Library/Caches/Homebrew".freeze
+
+        # Creating the basic directory structure needed for Homebrew
+        directories = ["bin", "etc", "include", "lib", "sbin", "share", "var", "opt",
+                        "share/zsh", "share/zsh/site-functions",
+                        "var/homebrew", "var/homebrew/linked",
+                        "Cellar", "Caskroom", "Homebrew", "Frameworks"
+                      ].freeze
+        directories.each do |dir|
+          directory "/usr/local/#{dir}" do
+            mode "0755"
+            owner new_resource.user
+            group "admin"
+            action :create
+          end
+        end
+
+        user_directories = ["#{USER_HOME}", "#{USER_HOME}/Library",
+                            "#{USER_HOME}/Library/Caches", "#{USER_HOME}/Library/Caches/Homebrew"
+        ]
+        user_directories.each do |dir|
+          directory "#{dir}" do
+            mode "0755"
+            owner new_resource.user
+            group "admin"
+            action :create
+          end
+        end
+
+        if new_resource.xcode_tools_url
+          dmg_package new_resource.xcode_tools_pkg_name do
+            source new_resource.xcode_tools_url
+            type "pkg"
+          end
+        else
+          build_essential "install Command Line Tools for Xcode" do
+            action :upgrade
+          end
+        end
+
+        script "Download and unpack Homebrew" do
+          interpreter "bash"
+          cwd "/usr/local/Homebrew"
+          code <<-CODEBLOCK
+            git init -q
+            curl #{new_resource.brew_source_url} -o brew-master.zip
+            unzip brew-master.zip -d /usr/local/Homebrew
+            rm /usr/local/Homebrew/brew-master.zip
```

Comment:
`tar` on macOS can handle zip files and can read from stdin, so this could be done as one command rather than 3. Here's an example of me trying this locally:

```
Petes-MBP:tmp pete$ mkdir -p brew
Petes-MBP:tmp pete$ cd brew/
Petes-MBP:brew pete$ curl "https://codeload.github.com/Homebrew/brew/zip/master" | tar -x -C .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3274k    0 3274k    0     0  2524k      0 --:--:--  0:00:01 --:--:-- 2524k
Petes-MBP:brew pete$ ls -l
total 0
drwxr-xr-x  19 pete  wheel  608 Dec 10 11:15 brew-master
```

Reply:
We should probably convert the download over to remote_file to handle it in a native chef way and download that file to our file cache dir

Reply:
got it

Reply:
OK, updated it to use remote_file to get the zip file. Testing now.